marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -673,16 +674,42 @@ instance Rewindable UtxoHandle where
 -- For resuming we need to provide a list of points where we can resume from.
 instance Resumable UtxoHandle where
   resumeFromStorage (UtxoHandle c _ _) = do
-    chainPoints <- fmap (uncurry C.ChainPoint) <$>
-            SQL.query c
-                [r|SELECT slotNo, blockHash
-                   FROM unspent_transactions
-                   ORDER BY slotNo DESC|] ()
+    chainPoints <- resumeHelper c
     -- The ordering here matters. The node will try to find the first point in the
     -- ledger, then move to the next and so on, so we will send the latest point
     -- first.
     pure $ chainPoints ++ [C.ChainPointAtGenesis]
 
+-- Add paination to resume
+-- Main reason for adding this is to protect against OOM
+-- TODO use withAsync to spread the load
+resumeHelper :: SQL.Connection -> IO [C.ChainPoint]
+resumeHelper c =
+  let
+    limit :: Int = 216000
+    traceName :: Text.Text="marconi-utxo-resume"
+    helper :: Int -> [C.ChainPoint] -> IO [C.ChainPoint]
+    helper offset tally =  do
+      d <- defaultConfigStdout
+      cps <- fmap (uncurry C.ChainPoint) <$>
+        SQL.query c [r|SELECT DISTINCT slotNo, blockHash
+                   FROM unspent_transactions
+                   ORDER BY slotNo DESC
+                   LIMIT ? OFFSET ? |] (limit, offset)
+      case cps of
+        [] -> withTrace d traceName $ \trace -> do
+          logInfo trace $ renderStrict $
+            layoutPretty defaultLayoutOptions $
+            "Resume is complted."
+          pure tally
+        cps' -> withTrace d traceName $ \trace -> do
+          logInfo trace $ renderStrict $
+            layoutPretty defaultLayoutOptions $
+             "Resume is at offset: " <+> pretty offset
+          helper (limit + offset)(tally <> cps')
+  in helper 0 []
```

Comment:
I think that should be part of another PR, no?

Reply:
I have mixed feeling about this. 
On the one hand, I could not complete the requirements without this fix for marconi-chain-index crashed with OOM on every attempt to resume from the last crash.  A complete mainnet macaroni-chain-index execution and showing that there are no duplicate records in the `spent` table were part of the acceptance criteria.   Starting marconi from Genesis and hoping all goes without another incident, didn't seem like the right approach either.
On the other hand, this fix is incomplete and requires more work. It takes a long time for `resume` to catch up, see below for some data.  
So my thoughts are:

Push the PR with this fix so that we can continue with tests on mainnet
Work on the next PR to do the `resume` concurrently, this will be a quick PR
Discuss the `resume` within our team and make it better as I think it needs some rethinking 

However, I have no objection to taking it out and doing separate PR for it

============= here are the logs showing how long it took to resume===============
```
[marconi-utxo-resume:Info:1] [2023-04-08 17:29:53.85 UTC] Resume is at offset:  0
[marconi-utxo-resume:Info:1] [2023-04-08 17:31:53.48 UTC] Resume is at offset:  216000
[marconi-utxo-resume:Info:1] [2023-04-08 17:33:58.57 UTC] Resume is at offset:  432000
[marconi-utxo-resume:Info:1] [2023-04-08 17:36:04.26 UTC] Resume is at offset:  648000
[marconi-utxo-resume:Info:1] [2023-04-08 17:38:10.28 UTC] Resume is at offset:  864000
[marconi-utxo-resume:Info:1] [2023-04-08 17:40:19.56 UTC] Resume is at offset:  1080000
[marconi-utxo-resume:Info:1] [2023-04-08 17:42:31.97 UTC] Resume is at offset:  1296000
[marconi-utxo-resume:Info:1] [2023-04-08 17:44:44.95 UTC] Resume is at offset:  1512000
[marconi-utxo-resume:Info:1] [2023-04-08 17:46:58.12 UTC] Resume is at offset:  1728000
[marconi-utxo-resume:Info:1] [2023-04-08 17:49:10.45 UTC] Resume is at offset:  1944000
[marconi-utxo-resume:Info:1] [2023-04-08 17:51:21.41 UTC] Resume is at offset:  2160000
[marconi-utxo-resume:Info:1] [2023-04-08 17:53:32.81 UTC] Resume is at offset:  2376000
[marconi-utxo-resume:Info:1] [2023-04-08 17:55:46.75 UTC] Resume is at offset:  2592000
[marconi-utxo-resume:Info:1] [2023-04-08 17:58:00.92 UTC] Resume is at offset:  2808000
[marconi-utxo-resume:Info:1] [2023-04-08 18:00:13.64 UTC] Resume is at offset:  3024000
[marconi-utxo-resume:Info:1] [2023-04-08 18:02:27.44 UTC] Resume is at offset:  3240000
[marconi-utxo-resume:Info:1] [2023-04-08 18:04:43.20 UTC] Resume is at offset:  3456000
[marconi-utxo-resume:Info:1] [2023-04-08 18:06:56.71 UTC] Resume is at offset:  3672000
[marconi-utxo-resume:Info:1] [2023-04-08 18:09:09.65 UTC] Resume is at offset:  3888000
[marconi-utxo-resume:Info:1] [2023-04-08 18:11:25.14 UTC] Resume is at offset:  4104000
[marconi-utxo-resume:Info:1] [2023-04-08 18:13:40.44 UTC] Resume is at offset:  4320000
[marconi-utxo-resume:Info:1] [2023-04-08 18:15:55.48 UTC] Resume is at offset:  4536000
[marconi-utxo-resume:Info:1] [2023-04-08 18:18:13.83 UTC] Resume is at offset:  4752000
[marconi-utxo-resume:Info:1] [2023-04-08 18:20:32.94 UTC] Resume is at offset:  4968000
[marconi-utxo-resume:Info:1] [2023-04-08 18:22:53.59 UTC] Resume is at offset:  5184000
[marconi-utxo-resume:Info:1] [2023-04-08 18:25:10.52 UTC] Resume is completed.
```

Reply:
There is additional info regarding `resume` issue on [slack](https://input-output-rnd.slack.com/archives/C029B8E2RHA/p1680886760652209)

Reply:
> Push the PR with this fix so that we can continue with tests on mainnet
Work on the next PR to do the resume concurrently, this will be a quick PR
Discuss the resume within our team and make it better as I think it needs some rethinking

Alright, let's do that